stock_serial_no_auto_reserve/models/stock_move.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2024 Sunflower IT (http://sunflowerweb.nl)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _action_assign(self):
+        """Dont auto-reserve products with serial number.
+        People will have to manually reserve. With regular Odoo
+        manual stock.move.line creation this is not possible, people
+        will have to use the tag icon from stock_quant_manual_assign"""
+        self_not_serial = self.filtered(
+            lambda m: m.product_id.tracking != "serial"
+            or m.product_id.serial_auto_reserve
+        )
+        if self_not_serial:
+            result = super(
+                StockMove, self_not_serial.with_context(from_assign=True)
+            )._action_assign()
+            move.picking_id._compute_is_available()
```

Comment:
What I meant was:

- `self_not_serial` is all the moves **except** those that have products that are tracked by serial number. On these, we call super.
- We then also calculate `self_with_serial` that has all the other moves, eg the serially tracked ones. These are the ones that skip reservation. On those, instead of reservation, we calculate an is_available boolean. This boolean should be a regular field defined on `stock.move`.
- We then define `is_available` as a stored computed field on `stock.picking`, and let it depend on move_ids_without_package.state as well as move_ids_without_package.is_available. The computation will then be along the lines of "loop all the moves, check if state is assigned, if not, check if state is waiting and is_available = true. If one of these conditions is true for all move lines, then is_available for the picking should turn out True. Otherwise, False.

What you're doing here is a different approach which might in theory work, but in its current form I'm seeing some possible flaws:

- The call here to the compute function is only called when `self_not_serial` has values, so it will not be called for those pickings that have serially-tracked products only.
- I don't see `move` being defined anywhere as a variable

Reply:
I would say whenever action_assign is called on any move that belongs to a certain picking, we always need to recalculate is_available boolean of that picking, because it may have changed. Depending on the calculation we make it True or False, but we always decide something.